mentat/code_context.py

```diff
@@ -63,61 +55,39 @@ def _get_all_features(
                 all_features.append(full_feature)
             else:
                 _split_features = split_file_into_intervals(
-                    git_root,
+                    path,
                     full_feature,
                     user_features=[f for f in include_files.values() if f.path == path],
                 )
                 all_features += _split_features
         else:
-            _feature = CodeFeature(
-                abs_path, level=level, diff=diff_target, user_included=user_included
-            )
+            _feature = CodeFeature(path, level=level, diff=diff_target, user_included=user_included)
             all_features.append(_feature)
 
     return all_features
 
 
 class CodeContext:
-    include_files: dict[Path, CodeFeature]
-    ignore_files: set[Path]
-    diff_context: DiffContext
-    code_map: bool = True
-    features: list[CodeFeature] = []
-    diff: Optional[str] = None
-    pr_diff: Optional[str] = None
-
-    def __init__(
-        self,
-        stream: SessionStream,
-        git_root: Path,
-        diff: Optional[str] = None,
-        pr_diff: Optional[str] = None,
-    ):
-        self.diff = diff
-        self.pr_diff = pr_diff
-        self.diff_context = DiffContext(stream, git_root, self.diff, self.pr_diff)
+    def __init__(self, ignore_patterns: Iterable[Path] = [], diff_context: DiffContext | None = None):
+        self.ignore_patterns = set(ignore_patterns)
+        self.diff_context = diff_context
+
+        # try:
+        #     git_root = get_shared_git_root_for_paths(paths)
+        #     self.diff_context = DiffContext(stream, git_root, self.diff, self.pr_diff)
+        # except (UserError, subprocess.CalledProcessError):
+        #     self.diff_context = None
+
         # TODO: This is a dict so we can quickly reference either a path (key)
         # or the CodeFeature (value) and its interval. Redundant.
-        self.include_files = {}
-        self.ignore_files = set()
-
-    def set_paths(
-        self,
-        paths: list[Path],
-        exclude_paths: list[Path],
-        ignore_paths: list[Path] = [],
-    ):
-        self.include_files, invalid_paths = get_include_files(paths, exclude_paths)
-        for invalid_path in invalid_paths:
-            print_invalid_path(invalid_path)
-        self.ignore_files = get_ignore_files(ignore_paths)
+        self.include_files: Dict[Path, CodeFeature] = dict()
+        self.features: List[CodeFeature] = list()
+        self.code_map = True
 
     def set_code_map(self):
-        session_context = SESSION_CONTEXT.get()
-        config = session_context.config
-        stream = session_context.stream
+        ctx = SESSION_CONTEXT.get()
```

Comment:
Thanks for changing this. It's a lot cleaner. Though it makes it even harder to tell what depends on what. And I'd rather it wasn't done as part of this PR. It makes the total number of lines changed massive. And it makes it hard to notice the little bits of logic that may have changed anywhere `git_root` was used before.

Reply:
Yea that makes sense, I'll revert this